### DISCUSSION TITLE
what is graphql?

### DISCUSSION THREAD

**Discussion by ntdat104** [ntdat104/fb-app#4] [Q&A] [answered]
- [ ] What is GraphQL and why it is used?

- [ ] What is GraphQL vs REST API?

- [ ] Is GraphQL frontend or backend?

- [ ] What is GraphQL vs SQL?

- [ ] Is GraphQL just JSON?

- [ ] Is GraphQL only for SQL?

**Comment by datnt187302** [comment, marked as answer]
GraphQL is a query language for APIs and a runtime for fulfilling those queries with your existing data. GraphQL provides a complete and understandable description of the data in your API, gives clients the power to ask for exactly what they need and nothing more, makes it easier to evolve APIs over time, and enables powerful developer tools.
- https://graphql.org/

**Reply by ntdat104** [reply]
ok
